[18.0][MIG] purchase_sale_inter_company: Migration to 18.0

metaminux opened the pull request:
Standard migration

@Kev-Roche

chafique-delli committed:
[ADD] purchase_sale_inter_company module

chafique-delli committed:
add warehouse_id in the values for create sale order

chafique-delli committed:
refactor code after add dependency on account_invoice_inter_company

chafique-delli committed:
check product is intercompany

chafique-delli committed:
extend invoice_method field and add action_cancel method for purchase and auto_purchase_line_id field in sale_order_line

chafique-delli committed:
rename inter_company_po_to_so_view.xml to res_company_view.xml and add purchase_order_view.xml

chafique-delli committed:
use product_id_change_with_wh method for return sale line data

chafique-delli committed:
fix README file

chafique-delli committed:
remove inter company user

chafique-delli committed:
remove unnecessary dependencies in openerp.py and update of the unit price in purchase line when validating the sale order

chafique-delli committed:
add demo data and tests

chafique-delli committed:
fix partner_shipping_id in _prepare_sale_order_data method

chafique-delli committed:
add force_company in context when create inter company sale oder

sebastienbeau committed:
[FIX] fix setting the force company pass the cest_company as an id in order to load it with the correct context

chafique-delli committed:
replace sale_line_vals by sale_line_data

chafique-delli committed:
fix tests

chafique-delli committed:
fix following @elicoidal's comments and add french translation

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

chafique-delli committed:
fix domain for search users in _check_intercompany_product method

chafique-delli committed:
fix multi-company access rights issue

OCA Transbot updated translations from Transifex

carlosdauden committed:
[MIG] purchase_sale_inter_company: Migration to 11.0 (from 8.0)

carlosdauden committed:
[IMP] purchase_sale_inter_company: Recover intercompany user

pedrobaeza committed:
[FIX] purchase_sale_inter_company: Transfer dropship pickings correctly

carlosdauden committed:
[IMP] purchase_sale_inter_company: Better relation between docs. Translate.

yajo committed:
[FIX] purchase_sale_inter_company: No buggy demo data

Make tests compatible with https://github.com/OCA/multi-company/pull/110, for the same reason as the specified in that link.

chienandalu committed:
[FIX] tests and other issues

mourad-ehm committed:
[MIG] purchase_sale_inter_company: Migration to 12.0

mourad-ehm committed:
[FIX] OCA Review

mourad-ehm committed:
[FIX] ref uom.product_uom_unit

mourad-ehm committed:
[FIX] import Warning

mourad-ehm committed:
[FIX] import Warning (To squash)

mourad-ehm committed:
[FIX] demo test data

mourad-ehm committed:
[FIX] add demo product

mourad-ehm committed:
[FIX] product_qty

mourad-ehm committed:
[FIX] unit test

mourad-ehm committed:
[FIX] pylint

mourad-ehm committed:
[FIX] set readonly = false for  all res_config_settings fields

bealdav committed:
IMP purch_sale_inter_cpny: layout settings

[UPD] Update purchase_sale_inter_company.pot

eLBati committed:
Added translation using Weblate (Italian)

AdriaGForgeFlow committed:
[IMP] purchase_sale_inter_company: black, isort

AdriaGForgeFlow committed:
[13.0][MIG] purchase_sale_inter_company

rename intercompany_user_id to intercompany_sale_user_id for compatibility with account_invoice_inter_company

[IMP] res_config_settings view

[UPD] Update purchase_sale_inter_company.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/

dong-z committed:
Added translation using Weblate (Chinese (Simplified))

AaronHForgeFlow committed:
[FIX]purchase_sale_inter_company tests
* product multi company has company because of the default method

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/

AaronHForgeFlow committed:
[IMP] purchase_sale_inter_company
* move demo data to SetupClass

purchase_sale_inter_company 13.0.1.0.1

chienandalu committed:
[FIX] purchase_sale_inter_company: commitment_date

The field was renamed from `requested_date` to `commitment_date`

purchase_sale_inter_company 13.0.1.0.2

Commit:
[MIG] purchase_sale_inter_company: Migration to 14.0

[UPD] Update purchase_sale_inter_company.pot

[IMP] update dotfiles [ci skip]

enriquemartin committed:
[FIX] purchase_sale_inter_company: propagate correct uom to sale.order.line

Commit:
[13.0] formatting

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (27 of 27 strings)

Translation: multi-company-13.0/multi-company-13.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-purchase_sale_inter_company/es/

purchase_sale_inter_company 13.0.1.0.4

kittiu committed:
[13.0][FIX] purchase_sale_inter_company, PO->SO error when has line_section

purchase_sale_inter_company 13.0.1.0.5

Commit:
[IMP] purchase_sale_inter_company: black, isort, prettier

Kev-Roche committed:
[FIX] purchase_sale_inter_company: fix new sale date_order

purchase_sale_inter_company 14.0.1.0.1

DhavalTforgeFlow committed:
[MIG] Migration to version 15.0

DhavalTforgeFlow committed:
[MIG] purchase_sale_inter_company: Migration to 15.0

yankinmax committed:
[ADD] purchase_sale_inter_company:

This is the splitted module from original to remove dependency on stock

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

Pricelist currency should be same with PO/SO document
If corresponding pricelist currency is not the same find or create propriate pricelist

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

Company contacts can be Vendors to create SO from PO

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

If purchase_exception or sale_exception modules are installed
we have an issue in tests, because order's confirm methods won't be called.
Possible fix is to set ignore_exception=True on orders in tests.
Exception check could be skipped by setting check_exception=False,
as mentioned in base_exception, but in fact it doesn't work.

yankinmax committed:
[15.0][IMP] purchase_sale_inter_company:

Pricelist is set with the company

JasminSForgeFlow committed:
[15.0][IMP] purchase_sale_inter_company:
Update v14 latest changes related to test case failing

[UPD] Update purchase_sale_inter_company.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-15.0/multi-company-15.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-15-0/multi-company-15-0-purchase_sale_inter_company/

MiquelRForgeFlow committed:
[FIX] purchase_sale_inter_company: typo in readme

[UPD] README.rst

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 75.0% (15 of 20 strings)

Translation: multi-company-15.0/multi-company-15.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-15-0/multi-company-15-0-purchase_sale_inter_company/it/

victoralmau committed:
[MIG] purchase_sale_inter_company: Migration to 16.0

TT45184

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (20 of 20 strings)

Translation: multi-company-16.0/multi-company-16.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-purchase_sale_inter_company/es/

chienandalu committed:
[FIX] purchase_sale_inter_company: sync confirmed unlocked purchase updates

When the purchase user updates the PO after the SO is generated and
always if both are confirmed and unlocked, we sync the new lines to update the demand.

TT46773

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (25 of 25 strings)

Translation: multi-company-16.0/multi-company-16.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-purchase_sale_inter_company/es/

JordiBForgeFlow committed:
[IMP] purchase_sale_inter_company: add method _check_intercompany_product
to purchase order line for better extensibility

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (25 of 25 strings)

Translation: multi-company-16.0/multi-company-16.0-purchase_sale_inter_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-purchase_sale_inter_company/it/

cuongnmtm committed:
[IMP] purchase_sale_inter_company: pre-commit auto fixes

cuongnmtm committed:
[MIG] purchase_sale_inter_company: Migration to 17.0

JordiMForgeFlow committed:
[IMP] purchase_sale_inter_company: return created SO

Commit:
[UPD] Update purchase_sale_inter_company.pot

OCA-git-bot committed:
[BOT] post-merge updates

metaminux commented:
Depends on:

- [x] #757

legalsylvain commented:
/ocabot migration purchase_sale_inter_company

metaminux commented:
> /ocabot migration purchase_sale_inter_company

Hello @legalsylvain

OK, seems this task is "broken"... always trying to update #782 instead of #705 (I just received an email telling that #782 has just been created...)
Maybe we need to reopen #782 and close #705 instead ?

np-oerp commented:
Hi @OCA-git-bot ,

Just wanted to know are you waiting for anything further on this PR?. If everything looks good from your side, can you please merge it into 18.0 branch as well ?

Thanks in advance!

carlos-lopez-tecnativa committed:
[MIG] purchase_sale_inter_company: allow reuse of utilities in another test
- Function _approve_po: allow passing purchase_to_approve
- Function _create_purchase_order: allow passing products

metaminux commented:
Standard migration

Includes:

-  [ ] #871 

I noticed #860 is also migrating `purchase_sale_inter_company` to 18.0, but is not declared in #705 (nor #782)

This one includes last changes from 17.0 and CI is green :
 - 8ff490bae577ef9570e1f513a85fdfe3c12575f1

CC @HviorForgeFlow @carlos-lopez-tecnativa

carlos-lopez-tecnativa commented on purchase_sale_inter_company/models/account_move.py:
```diff
@@ -0,0 +1,32 @@
+        if orders:
+            message = _("This vendor bill is related with: {}").format(
+                ",".join([o._get_html_link(o.name) for o in orders])
+            )
```
```suggestion
            message = self.env._("This vendor bill is related with: %(origin)s", 
                origin=",".join([o._get_html_link(o.name) for o in orders])
            )
```

metaminux replied:
Done ! Thanks.

carlos-lopez-tecnativa commented on purchase_sale_inter_company/models/purchase_order.py:
```diff
@@ -0,0 +1,188 @@
+            or dest_company.currency_id.id
+        ):
+            raise UserError(
+                _(
```
```suggestion
                self.env._(
```

metaminux replied:
Done !

carlos-lopez-tecnativa commented on purchase_sale_inter_company/models/purchase_order.py:
```diff
@@ -0,0 +1,188 @@
+        )
+        for so in sale_orders:
+            if so.state not in ["draft", "sent", "cancel"]:
+                raise UserError(_("You can't cancel an order that is %s") % so.state)
```
And check the remaining occurrences
```suggestion
                raise UserError(self.env._("You can't cancel an order that is %s") % so.state)
```

metaminux replied:
Done for all occurences, thanks

carlos-lopez-tecnativa commented on purchase_sale_inter_company/tests/test_inter_company_purchase_sale.py:
```diff
@@ -0,0 +1,396 @@
+                "name": "test_account_income",
+                "code": "987",
+                "account_type": "income",
+                "company_ids": [(6, 0, [cls.company_b.id])],
```
Please use [Command](https://github.com/odoo/odoo/blob/028e7228cb830e47a9726bef4c82793ba4590cd5/odoo/fields.py#L4237-L4266
), it is more readable

```suggestion
                "company_ids": [Command.set([cls.company_b.id])],
```

metaminux replied:
Done !

carlos-lopez-tecnativa commented on purchase_sale_inter_company/tests/test_inter_company_purchase_sale.py:
```diff
@@ -0,0 +1,396 @@
+                "code": "765",
+                "account_type": "expense",
+                "reconcile": True,
+                "company_ids": [(6, 0, [cls.company_a.id])],
```
```suggestion
                "company_ids": [Command.set([cls.company_a.id])],
```

metaminux replied:
Done !

carlos-lopez-tecnativa commented on purchase_sale_inter_company/tests/test_inter_company_purchase_sale.py:
```diff
@@ -0,0 +1,396 @@
+            self.product.company_ids = [(6, 0, [self.company_a.id])]
+        self.product.company_id = self.company_a
+        with self.assertRaises(UserError):
+            self._approve_po()
```
Personally, I prefer using `assertRaisesRegex` and specifying the expected message. Could you please rework all `assertRaises` to use `assertRaisesRegex`?

metaminux replied:
All done, thanks !

carlos-lopez-tecnativa reviewed (changes requested):
LGTM, just a minor changes that I hope you can update to improve the code.

carlos-lopez-tecnativa commented on purchase_sale_inter_company/models/purchase_order.py:
```diff
@@ -0,0 +1,190 @@
+            if so.state not in ["draft", "sent", "cancel"]:
+                raise UserError(
+                    self.env._("You can't cancel an order that is %s") % so.state
+                )
```
In the translation function, you can pass the arguments as parameters.

```suggestion
                raise UserError(
                    self.env._("You can't cancel an order that is %s", so.state)
                )
```

carlos-lopez-tecnativa commented on purchase_sale_inter_company/models/res_config.py:
```diff
@@ -0,0 +1,34 @@
+
+    so_from_po = fields.Boolean(
+        related="company_id.so_from_po",
+        string="Create Sale Orders when buying to this company",
+        help="Generate a Sale Order when a Purchase Order with this company "
+        "as supplier is created.\n The intercompany user must at least be "
+        "Sale User.",
```
In related fields, the attributes are inherited, so I think it is not necessary to redefine these attributes again.
```suggestion
```

carlos-lopez-tecnativa commented on purchase_sale_inter_company/models/res_config.py:
```diff
@@ -0,0 +1,34 @@
+    )
+    sale_auto_validation = fields.Boolean(
+        related="company_id.sale_auto_validation",
+        string="Sale Orders Auto Validation",
+        help="When a Sale Order is created by a multi company rule for "
+        "this company, it will automatically validate it.",
```
```suggestion
```

carlos-lopez-tecnativa commented on purchase_sale_inter_company/models/res_config.py:
```diff
@@ -0,0 +1,34 @@
+    intercompany_sale_user_id = fields.Many2one(
+        comodel_name="res.users",
+        related="company_id.intercompany_sale_user_id",
+        string="Intercompany Sale User",
+        help="User used to create the sales order arising from a purchase "
+        "order in another company.",
```
```suggestion
```

carlos-lopez-tecnativa reviewed (approved):
LGTM
Non-blocking comment, but if you apply it, that would be great.

TT57264
ping @pedrobaeza

metaminux commented:
All suggestions applied, thank you @carlos-lopez-tecnativa

pedrobaeza commented on purchase_sale_inter_company/models/res_config.py:
```diff
@@ -11,24 +11,14 @@ class InterCompanyRulesConfig(models.TransientModel):
     so_from_po = fields.Boolean(
         related="company_id.so_from_po",
-        string="Create Sale Orders when buying to this company",
-        help="Generate a Sale Order when a Purchase Order with this company "
```
These help fields are lost in the screen. Can you recover them?

metaminux committed:
[MIG] purchase_sale_inter_company: Migration to 18.0

metaminux commented:
Hi @carlos-lopez-tecnativa and @pedrobaeza 

As #871 seems not really useful in 17.0, I dropped it in this migration.

I also recovered `help` and `string` fields as suggested in `res.config.settings`

CI is green...

pedrobaeza reviewed (approved):
/ocabot merge nobump
